core/src/main/java/com/linecorp/armeria/common/stream/StreamMessageUtil.java

```diff
@@ -28,29 +26,9 @@
 
 final class StreamMessageUtil {
 
-    public static final SubscriptionOption[] EMPTY_OPTIONS = {};
-
-    static boolean containsWithPooledObjects(SubscriptionOption... options) {
-        requireNonNull(options, "options");
-        for (SubscriptionOption option : options) {
-            if (option == SubscriptionOption.WITH_POOLED_OBJECTS) {
-                return true;
-            }
-        }
-
-        return false;
-    }
-
-    static boolean containsNotifyCancellation(SubscriptionOption... options) {
-        requireNonNull(options, "options");
-        for (SubscriptionOption option : options) {
-            if (option == SubscriptionOption.NOTIFY_CANCELLATION) {
-                return true;
-            }
-        }
-
-        return false;
-    }
+    static final SubscriptionOption[] CANCELLATION_OPTION = { SubscriptionOption.NOTIFY_CANCELLATION };
```

Comment:
I tried it at first. 
However, `StreamMessageUtil` depends on `AbortingSubscriber` which is a package-private class.
`AbortingSubscriber` depends on `AbortedStreamException.INSTANCE` which is also package-private.
Because of the mess dependencies, I reverted it. If you think it is still worth merging them, let try it again. :-)

Reply:
How about moving `CANCELLATION_OPTION` and `CANCELLATION_AND_POOLED_OPTIONS` at least?
Because it seems a bit weird that `EMPTY_OPTIONS` and `POOLED_OBJECTS` are there but the rest are here. 🤔